File: Dot examples/ASP Book Examples/Book Examples/Chapter34/vb/App_Code/Entities/Listing.vb
Imports System
Imports System.Linq
Imports System.Data.Linq.Mapping
Imports System.Collections.Generic

Partial Public Class Listing
     Inherits EntityBase(Of Listing)


    Public Shared Function SelectByChapterId(ByVal chapterId As Integer) As IEnumerable(Of EnTry)
        Return Table _
            .Where(Function(l) l.ChapterId = chapterId) _
            .OrderBy(Function(l) l.OrderNumber) _
            .Join(Entry.Table, Function(l) l.EntryId, Function(e) e.Id, Function(l, e) e)


    End Function

    Protected Overrides Sub Validate()
    End Sub

End Class

File: Dot examples/ASP Book Examples/Book Examples/Chapter34/cs/App_Code/Entities/Listing.cs
using System;
using System.Linq;
using System.Data.Linq.Mapping;
using System.Collections.Generic;

public partial class Listing: EntityBase<Listing>
{
    

    public static IEnumerable<Entry> SelectByChapterId(int chapterId)
    {
        return Table
            .Where( l => l.ChapterId == chapterId )
            .OrderBy( l => l.OrderNumber )
            .Join(Entry.Table, l => l.EntryId, e => e.Id, (l, e) => e);
            
            
    }

    protected override void Validate()
    {
    }

}
